### PR TITLE
Phase 3

### PR DESCRIPTION
**Pull request by mhoobler**
- [x] As a User, I can see the item info on the item's page
  - Implement the design of the item page so the user can see the item's image, name, description, etc.
- [x] As a User, I can see the item's rating, and the coresponding stars on the item cards on the home page
  - Use the component created on the previous task to render rating stars on the item cards.
- [x] As a User, I can select the number of items I can add to my cart.
  - Add a way for the user to be able to select the number of items to add to the cart. Make sure the user can't add more items than there are in stock.
- [x] As a User, I can see an error message if I try to add more item's to my cart than are in stock.
- [x] As a User, I should see an error message if I try to add an invalid item count to my cart
  - Adding something like a negative number, or 0, should display an error message to the user.
- [x] As a User, I should no longer see an error message if I add a valid amount of items to my cart
  - If the user sees an error message, and the proceeds to add a valid amount of items to their cart, the error message should be removed.

### CODE REVIEW

**Comment by mhoobler** [comment]
The "already in cart error" just exists statically on the page currently since there's no cart functionality yet. Won't be hard making it dynamic once the cart is implemented.

File: client/src/pages/Deals.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+import React from "react";
 
 import ItemContainer from "../components/ItemContainer";
-import useGetItems from "../utils/useGetItems";
+import useGetList from "../utils/useGetList";
 
 const Deals: React.FC = () => {
   const query: ListQuery = { isOnSale: true };
-  const itemsList = useGetItems(JSON.stringify(query));
+  const itemsList = useGetList(JSON.stringify(query));
```
Comment (pchertude34) [inline client/src/pages/Deals.tsx:8]: I love the custom hooks for fetching the data. Super cool. I didn't even think to do this when I built the app out myself! But you really don't need the `JSON.stringify` here. If you remove the `JSON.stringify` you will have to make some changes in your hook to not run infinite queries. Specifically in the props definition and the useEffect dependencies array.

File: client/src/pages/Home.tsx [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-import useGetItems from "../utils/useGetItems";
+import useGetList from "../utils/useGetList";
 
 import "./styles/Home.css";
 
 const Home: React.FC = () => {
   const [search, setSearch] = useState("");
   const query: ListQuery = { size: 29, q: search };
-  const itemsList = useGetItems(JSON.stringify(query));
+  const itemsList = useGetList(JSON.stringify(query));
```
Comment (pchertude34) [inline client/src/pages/Home.tsx:13]: Don't need the `JSON.stringify` here either

File: client/src/components/Price/index.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+import "./Price.css";
+
+type PriceProps = {
+  price: number;
+  isOnSale?: boolean;
+};
+
+const Price: React.FC<PriceProps> = ({ price, isOnSale }) => {
```
Comment (pchertude34) [inline client/src/components/Price/index.tsx:10]: This component is awesome!

File: client/src/pages/ItemPage.tsx [hunk 4]
```diff
@@ -1,14 +1,83 @@
...
+              <div className="item-rating">
+                {<Ratings rating={item.avgRating} />}
+              </div>
+
+              <div className="item-description">{item.description}</div>
+
+              <Price price={item.price} />
+
+              <QuantitySelect
+                handleChange={(n: number) => {
```
Comment (pchertude34) [inline client/src/pages/ItemPage.tsx:54]: Move this out into a function on the component. Anytime that you have a function that is more than 1 or 2 lines it should not be inlined.

File: client/src/pages/Deals.tsx (hunk 1 above)
Comment (mhoobler) [reply client/src/pages/Deals.tsx:8]: Thanks, @maistho actually suggested it. I never felt like I ran into use case for a custom hook like that, but after using it here I definitely should've been using them more often.

File: client/src/components/Button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import "./Button.css";
+
+type ButtonProps = {
+  align?: "left" | "right";
+};
+
+const Button: React.FC<ButtonProps> = ({ align, children }) => {
+  return (
+    <div className={`button-container ${align ? `align-${align}` : ''}`}>
```
[the remark this answers is not in the document]
Comment (mhoobler) [reply client/src/components/Button/index.tsx:11]: I think that's more of an issue with my naming conventions. It's definitely confusing, but basically all that component does is style things to look like that "orange button" on the example design.

**Pull request merged**